### PR TITLE
Check a Comparable type can be deserialized as a Scalar before writing it as a Scalar type, #438

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/wire/serializable/MonetaTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @Test
+    public void monetary() {
+        SortedSet<NonScalarComparable> set = new TreeSet<>();
+        for (String curr : "GBP,USD,EUR,AUD".split(","))
+            set.add(new NonScalarComparable(Currency.getInstance(curr)));
+
+        Wire wire = new TextWire(Bytes.allocateElasticOnHeap());
+        wire.write("currencies")
+                .object(set);
+//        System.out.println(wire);
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/wire/serializable/MonetaTest.java]: can we get rid of commented-out code?

File: src/test/java/net/openhft/chronicle/wire/Issue341Test.java [hunk 2]
```diff
@@ -26,8 +26,8 @@ public Issue341Test(WireType wireType) {
     public static Collection<Object[]> data() {
         return Arrays.asList(new Object[][]{
                 new Object[]{WireType.TEXT},
-                new Object[]{WireType.JSON},
-                new Object[]{WireType.BINARY},
+                new Object[]{WireType.YAML},
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/wire/Issue341Test.java:35]: why change?

File: src/test/java/net/openhft/chronicle/wire/serializable/MonetaTest.java (hunk 1 above)
Comment (peter-lawrey) [reply src/test/java/net/openhft/chronicle/wire/serializable/MonetaTest.java]: Removed

File: src/test/java/net/openhft/chronicle/wire/Issue341Test.java (hunk 2 above)
Comment (peter-lawrey) [reply src/test/java/net/openhft/chronicle/wire/Issue341Test.java:35]: Added JSON_ONLY. Had dropped BINARY as have BINARY_LIGHT

**Pull request merged**